pages/blog/whats-new-in-graphql-codegen-v2.mdx

```diff
@@ -0,0 +1,135 @@
+
+[`n1ru4l`](https://github.com/n1ru4l) picked that up and [fixed the bug](https://github.com/dotansimha/graphql-code-generator/pull/6184) and added support for better handling of these kind of cases.
+
+The new configuration flag (`inlineFragmentTypes: inline`) is used by default, and generates a more accurate types, without introducing breaking changes!
+
+> All your types are still fully compatible with `v1` of the tool. You can set `inlineFragmentTypes: combine` if you prefer to keep the old behavior.
+
+## Removal of deprecations
+
+We also used the need for a major version to remove a few deprecations from the codebase.
+
+#### `typescript-resolvers`
+
+In `typescript-resolvers`, we had a generated signature for `IResolvers` and `IDirectiveResolvers` (which were deprecated 2 years ago), `v2` of this plugin is removing these proxy types.
+
+Also, `noSchemaStitching` flag is now set to `true` by default, so the generated resolvers signature is simpler and matches the needs of most projects.
+
+> If you are using Schema Stitching, you can set `noSchemaStitching: false` to keep the old behavior.
+
+#### `typescript-compatibility`
+
+The `typescript-compatibility` plugin was created to make it easier to migrate from v0 to v1 of codegen, and not being actively developed for a few years now.
+
+With this release, we no longer support or develop `typescript-compatibility` plugin, so if you are using it, please consider to migrate your usage and types to the new form.
+
+## New TypeScript Plugin for Operations!
+
+A month ago, [`Maël Nison`](https://github.com/arcanis) reached out to us with [a new concept](https://github.com/arcanis/graphql-typescript-integration) for matching your actual GraphQL operation string and the generated TypeScript types.
+
+With a few adjustments, [`n1ru4l`](https://github.com/n1ru4l) managed to [turn that amazing idea into a plugin and a preset](https://github.com/dotansimha/graphql-code-generator/pull/6267) that extends the behavior of `typescript-operation` and `TypedDocumentNode`, and allow you to generate TypeScript types that magically matches your GraphQL query strings (this also required a few fixes in `graphql-tools`, so now the Loaders in graphql-tools are better!).
+
+#### So what does it means?
+
+Today, GraphQL Codegen scans your codebase and looks for all the operations that are being used by your components (or, from `.graphql` files), then it generates types and wrapped `TypedDocumentNode` for you.
+
+With this new plugin+preset, you can generate typings for your inline `gql` function usages, without having to manually specify import statements for the documents. All you need to do is import your `gql` function and run codegen in watch mode.
```

Comment:
Should you show before/after example?

Reply:
@n1ru4l can you please add an example, or maybe explain it in a better way?

Reply:
yes